pkg/security/secl/model/model.go

```diff
@@ -331,7 +335,8 @@ func (p *Process) GetPathResolutionError() string {
 
 // HasInterpreter returns whether the process uses an interpreter
 func (p *Process) HasInterpreter() bool {
-	return p.LinuxBinprm.FileEvent.Inode != 0 && p.LinuxBinprm.FileEvent.MountID != 0
+	// there can be an error case in which inode != 0 && mount_id == 0 && !fileless (if path resolution fails), but this boolean drops that case
+	return p.LinuxBinprm.FileEvent.Inode != 0 || p.LinuxBinprm.FileEvent.isFilelessExecution()
```

Comment:
Why not simply return p.LinuxBinprm.FileEvent.Inode != 0 ?
Here, we are saying that, if this is a fileless execution then we have an interpreter which doesn't sound good

Reply:
We're saying that if the linux_binprm struct is populated with either inode != 0 or it's fileless, then we have an interpreter. IIRC, the linux_binprm struct is not populated if there's no interpreter. I'm going to double check